src/org/loklak/http/ClientConnection.java

```diff
@@ -363,8 +364,9 @@ public static String getRedirect(String urlstring, boolean useAuthentication, bo
                     if (debugLog) {
                         DAO.log("Fetched Meta redirect URL for " + urlstring + " : " + metaURL);
                     }
-                    if (!metaURL.startsWith("http") && !urlstring.endsWith("/")) {
-                        return urlstring + "/" + metaURL;
+                    if (!metaURL.startsWith("http")) {
+                        URL u = new URL(new URL(urlstring), metaURL);
+                        return u.toString();
```

Comment:
Why shall we use url as string is converted to url and then to string.
URL object is used to manipulate url, get it's parts, etc. what is the use here?